services/datamanager/builtin/file_deletion.go

```diff
@@ -0,0 +1,111 @@
+package builtin
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.viam.com/rdk/logging"
+
+	"github.com/ricochet2200/go-disk-usage/du"
+	"go.viam.com/rdk/services/datamanager/datacapture"
+	"go.viam.com/rdk/services/datamanager/datasync"
+)
+
+// TODO change these values back to what they should be
+const (
+	fileDeletionThreshold    = .2
+	captureDirRatioThreshold = .1
+	n                        = 4
+)
+
+var errAtSizeThreshold = errors.New("capture dir is at correct size")
+
+func checkFileSystemStats(ctx context.Context, captureDirPath string, logger logging.Logger) (bool, error) {
+	usage := du.NewDiskUsage(captureDirPath)
+	usedSpace := usage.Usage()
+	if math.IsNaN(float64(usedSpace)) {
```

Comment:
When would this happen?

Reply:
I ran into this because we have a default data manager but this can happen if the capture dir does not exist for any reason.

Reply:
Agreed that it's good to guard against this if the capture directory gets deleted for some reason — can you add a comment? The default data manager should have capture disabled and thus not hit the polling code, unless I'm missing something.

Reply:
Why wouldn't the default data manager have capture enabled? I thought `capture_disabled = false` for the default data manager, which means we would always hit this for the default.

Reply:
I also thought capture is enabled by default for now, will add a comment.

Reply:
You're correct, capture is enabled but without any collectors by default. Thanks!